Rust Polars v0.41.2 build fails

### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

```
cargo build
```


### Log output

```shell
error[E0432]: unresolved import `regex`
  --> .cargo/registry/src/index.crates.io-6f17d22bba15001f/polars-plan-0.41.2/src/dsl/function_expr/strings.rs:11:5
   |
11 | use regex::{escape, Regex};
   |     ^^^^^ use of undeclared crate or module `regex`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `polars-plan` (lib) due to 1 previous error
```


### Issue description

Build Polars v0.41.2 without the `regex` feature

### Expected behavior

Builds successfully. 

### Installed versions

<details>

polars = { version = "0.41", features = ["abs", "propagate_nans", "rank", "dynamic_group_by", "performant", 
                                         "ndarray" ,"lazy", "dtype-time", "dtype-datetime", "parquet", "strings"] }

</details>

--- a/crates/polars-plan/src/dsl/function_expr/strings.rs
+++ b/crates/polars-plan/src/dsl/function_expr/strings.rs
@@ -8,6 +8,7 @@ use polars_core::chunked_array::temporal::validate_time_zone;
 use polars_core::utils::handle_casting_failures;
 #[cfg(feature = "dtype-struct")]
 use polars_utils::format_smartstring;
+#[cfg(feature = "regex")]
 use regex::{escape, Regex};
 #[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
@@ -15,6 +16,7 @@ use serde::{Deserialize, Serialize};
 use super::*;
 use crate::{map, map_as_slice};
 
+#[cfg(feature = "regex")]
 #[cfg(feature = "timezones")]
 static TZ_AWARE_RE: Lazy<Regex> =
     Lazy::new(|| Regex::new(r"(%z)|(%:z)|(%::z)|(%:::z)|(%#z)|(^%\+$)").unwrap());
@@ -648,6 +650,7 @@ fn to_datetime(
     let datetime_strings = &s[0].str()?;
     let ambiguous = &s[1].str()?;
     let tz_aware = match &options.format {
+        #[cfg(feature = "regex")]
         #[cfg(feature = "timezones")]
         Some(format) => TZ_AWARE_RE.is_match(format),
         _ => false,

--- a/crates/polars-plan/src/dsl/function_expr/strings.rs
+++ b/crates/polars-plan/src/dsl/function_expr/strings.rs
@@ -16,8 +16,7 @@ use serde::{Deserialize, Serialize};
 use super::*;
 use crate::{map, map_as_slice};
 
-#[cfg(feature = "regex")]
-#[cfg(feature = "timezones")]
+#[cfg(all(feature = "regex", feature = "timezones"))]
 static TZ_AWARE_RE: Lazy<Regex> =
     Lazy::new(|| Regex::new(r"(%z)|(%:z)|(%::z)|(%:::z)|(%#z)|(^%\+$)").unwrap());
 
@@ -650,8 +649,7 @@ fn to_datetime(
     let datetime_strings = &s[0].str()?;
     let ambiguous = &s[1].str()?;
     let tz_aware = match &options.format {
-        #[cfg(feature = "regex")]
-        #[cfg(feature = "timezones")]
+        #[cfg(all(feature = "regex", feature = "timezones"))]
         Some(format) => TZ_AWARE_RE.is_match(format),
         _ => false,
     };

